internal/provider/data_source_aci_netflow_monitor_policy.go

```diff
@@ -0,0 +1,202 @@
+						},
+						"netflow_record_policy_name": schema.StringAttribute{
+							Computed:            true,
+							MarkdownDescription: `Name.`,
+						},
+					},
+				},
+			},
+			"annotations": schema.SetNestedAttribute{
+				MarkdownDescription: ``,
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Computed:            true,
+							MarkdownDescription: `The key used to uniquely identify this configuration object.`,
+						},
+						"value": schema.StringAttribute{
+							Computed:            true,
+							MarkdownDescription: `The value of the property.`,
+						},
+					},
+				},
+			},
+			"tags": schema.SetNestedAttribute{
+				MarkdownDescription: ``,
+				Computed:            true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"key": schema.StringAttribute{
+							Computed:            true,
+							MarkdownDescription: `The key used to uniquely identify this configuration object.`,
+						},
+						"value": schema.StringAttribute{
+							Computed:            true,
+							MarkdownDescription: `The value of the property.`,
+						},
+					},
+				},
+			},
+		},
+	}
+	tflog.Debug(ctx, "End schema of datasource: aci_netflow_monitor_policy")
+}
+
+func (d *NetflowMonitorPolDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	tflog.Debug(ctx, "Start configure of datasource: aci_netflow_monitor_policy")
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	d.client = client
+	tflog.Debug(ctx, "End configure of datasource: aci_netflow_monitor_policy")
+}
+
+func (d *NetflowMonitorPolDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	tflog.Debug(ctx, "Start read of datasource: aci_netflow_monitor_policy")
+	var data *NetflowMonitorPolResourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	setNetflowMonitorPolId(ctx, data)
+
+	// Create a copy of the Id for when not found during getAndSetNetflowMonitorPolAttributes
+	cachedId := data.Id.ValueString()
+
+	tflog.Debug(ctx, fmt.Sprintf("Read of datasource aci_netflow_monitor_policy with id '%s'", data.Id.ValueString()))
+
+	getAndSetNetflowMonitorPolAttributes(ctx, &resp.Diagnostics, d.client, data)
+
+	if data.ParentDn.IsNull() || data.ParentDn.IsUnknown() {
+		data.ParentDn = basetypes.NewStringValue("uni/infra")
+	}
```

Comment:
Should this not be placed before the call of setNetflowMonitorPolId()? I think what would happen is that when the parent_dn is not provided it is set to null and then the getAndSetNetflowMonitorPolAttributes() will make a invalid rest request. How did you test this?